src/crucible/aws/neptune.clj

```diff
@@ -0,0 +1,23 @@
+(ns crucible.aws.neptune
+  "Resources in AWS::Neptune::*"
+  (:require [clojure.spec.alpha :as s]
+            [crucible.encoding.keys :refer [->key]]
+            [crucible.resources :refer [spec-or-ref defresource]]))
+
+;;; property specs shared among AWS::Neptune::* resources
+
+(def db-cluster-identifier-regex #"^[a-zA-Z][a-zA-Z0-9_\-]*[a-zA-Z0-9_]")
```

Comment:
+1 for completeness. Would love to see you try the 'two consecutive hyphens' restriction here :-)
We could easily add the 63 character limit though, but I'll take it as it is.

Reply:
Yea the consecutive hyphens thing is above my regex foo. Would be simple as another predicate as well as the character limit. I meant to add those but got forgotten in the shuffle.